### PR TITLE
feat(runner): Add monitorProviders support for runners

### PR DESCRIPTION
**Pull request by nalanj**
## Describe your changes

Adds new monitorProviders function to shared and utilizes it in runners to reload providers.yaml dynamically when a PROVIDERS_URL is set.

## Issue ticket number and link

https://linear.app/nango/issue/NAN-2105/dynamically-reload-providersyaml

## Checklist before requesting a review (skip if just adding/editing APIs & templates)
- [ ] I added tests, otherwise the reason is: 
- [ ] I added observability, otherwise the reason is:
- [ ] I added analytics, otherwise the reason is:

### CODE REVIEW

File: packages/shared/lib/services/providers.ts [hunk 1]
```diff
@@ -23,6 +24,48 @@ export function getProvider(providerName: string): Provider | null {
...
+export async function monitorProviders(): Promise<void> {
+    const providersUrl = process.env['PROVIDERS_URL'];
+
+    // fall back to standard disk loading if no URL is provided
+    if (!providersUrl) {
+        return;
+    }
+
+    const maybeReloadInterval = process.env['PROVIDERS_RELOAD_INTERVAL'];
+    const reloadInterval = parseInt(maybeReloadInterval ?? '') || 60000;
```
Comment (TBonnin) [inline packages/shared/lib/services/providers.ts]: if you use `parseEnvs(ENVS)` from the `utils` package you'll get all the this for free

File: packages/runner/lib/app.ts [hunk 2]
```diff
@@ -2,6 +2,9 @@ import './tracer.js';
 import { server } from './server.js';
 import { stringifyError } from '@nangohq/utils';
 import { logger } from './utils.js';
+import { monitorProviders } from '@nangohq/shared';
+
+await monitorProviders();
```
Comment (TBonnin) [inline packages/runner/lib/app.ts]: let's cleanup the interval timer in the `close` function below

**Review by TBonnin — APPROVED** [review]

File: packages/shared/lib/services/providers.ts (hunk 1 above)
Comment (nalanj) [reply packages/shared/lib/services/providers.ts]: Fixed in e70ed3f7

File: packages/runner/lib/app.ts (hunk 2 above)
Comment (nalanj) [reply packages/runner/lib/app.ts]: Fixed in e70ed3f7

**Pull request merged**